test/dynamics/signals/test_pulse_to_signals.py

```diff
@@ -157,6 +157,78 @@ def test_uneven_pulse_length(self):
         self.assertTrue(signals[0].carrier_freq == 2.0)
         self.assertTrue(signals[1].carrier_freq == 3.0)
 
+    def test_different_start_times(self):
```

Comment:
Not really in the scope of this PR, do these pulse schedules use `add_samples`?

Reply:
They do - actually all schedule -> signal conversions do: they are all initialized to empty `DiscreteSignal` instances and the samples for every pulse are appended via `add_samples`. Just a higher level test that depends on this functionality.